app/queries/balance_query.rb

```diff
@@ -15,25 +15,25 @@ def pending_balances
 
   def pending_debts
     @pending_debts ||= Dish
-      .joins(:order)
+      .joins(:order, :user_dishes)
       .where(
-        'orders.status = 1 AND orders.user_id != ? AND dishes.user_id = ?',
+        'orders.status = 1 AND orders.user_id != ? AND user_dishes.user_id = ?',
         user.id, user.id
       )
       .group('orders.user_id')
-      .pluck('orders.user_id, SUM(-price_cents - shipping_cents/dishes_count)')
+      .pluck('orders.user_id, SUM(-price_cents/user_dishes_count) - SUM(shipping_cents)/SUM(user_dishes_count)')
```

Comment:
I'm afraid that `SUM(shipping_cents)/SUM(user_dishes_count)` is not correct...
`SUM(user_dishes_count)` probably sums all user_dishes from all orders and `SUM(shipping_cents)` sums all shipping_cents from all orders...

I think we have to add `has_many :user_dishes through: :dishes` association to order and use it here.

Reply:
TODO, its for sure not correct

Reply:
Again, don't submit TODOs to a pull request. It means it's not finished.